process.ts

```diff
@@ -0,0 +1,96 @@
+import * as nodeProcess from 'node:process';
+
+import { Channel } from './channel';
+import { NotImplementedError } from './exceptions';
+import { formatter } from './format';
+import { Value } from './value';
+import { nullish } from './value/nullness';
+
+export type Pid = number;
+
+export type Env = Record<string, Value>;
+
+abstract class IOSpec {
+  public abstract nodeStdIO(): string;
+}
+
+export class Inherit extends IOSpec {
+  public nodeStdIO(): string {
+    return 'inherit';
+  }
+}
+
+export const INHERIT = new Inherit();
+
+export class Pipe extends IOSpec {
+  constructor(
+    public channel: Channel,
+    public toId: number | null = null,
+  ) {
+    super();
+  }
+
+  public nodeStdIO(): string {
+    return 'pipe';
+  }
+}
+
+export class Redirect extends IOSpec {
+  constructor(public channel: Channel) {
+    super();
+  }
+
+  public nodeStdIO(): string {
+    throw new NotImplementedError('Stream redirection');
+  }
+}
+
+export interface ProcessSpec {
```

Comment:
Can we iterate on this so it's closer to the process management we would expect in Rust or QT, in anticipation of migrating in that direction?